devdoc/channel.md

```diff
@@ -0,0 +1,267 @@
+# `channel` requirements
+
+## Overview
+
+`channel` is a module that allows the user to move data asynchronously from producer to consumer without having to block the thread.
+
+Consumers of data call `channel_pull` to register a `PULL_CALLBACK` function to be called when data becomes available.
+
+Producers of data call `channel_push` to notify the channel that data is available.
+
+`channel_pull` and `channel_push` can be called from different threads in any order.
+
+## `channel` state
+
+The state of the `channel` module consists of two components:
+- EPOCH: This is a monotonically increasing integer that is incremented every time an operation is completed or cancelled.
```

Comment:
>operation [](http://example.com/codeflow?start=86&length=9)

Now, if this "operation" is somehow linked to the completion of an API... then it might be fine. If it is the complete cycle of going through a push AND a pull, then I doubt this is fine.

Is the state incrementing at every API call? I guess the rest of the document will have answers.

And is this "operations" the same as the THANDLE(ASYNC_OP)?

Reply:
"operation" is "complete cycle of going through a push AND a pull"

Only need to increment epoch at the completion of the cycle